library/src/main/java/com/google/maps/android/clustering/view/DefaultClusterRenderer.java

```diff
@@ -290,7 +293,7 @@ public void run() {
             });
             renderTask.setProjection(projection);
             renderTask.setMapZoom(mMap.getCameraPosition().zoom);
-            new Thread(renderTask).start();
+            mExecutor.execute(renderTask);
```

Comment:
Just calling out for future reference that given this Executor is instantiated via `newSingleThreadExecutor()` we're changing to single-threading here. But, the entire implementation appears to be effectively single-threaded anyway given the `synchronized` block and `mViewModificationInProgress` variable which will stop any other RenderTasks from being executed until this one finishes. So :+1:

Reply:
That was my line of thinking exactly. There's only ever one render task in flight and potentially one queued up to run next.